### PR TITLE
 Add convenience container-runtime flag for kubeadm 

### PR DESCRIPTION
**Pull request by r2d4**
Depends on #2037 

To enable the cri-o runtime you may now just use

```
minikube start --container-runtime=cri-o --bootstrapper=kubeadm
```
or
```
minikube start --container-runtiume=crio --bootstrapper=kubeadm
```
or
```
minikube start --extra-config=kubelet.container-runtime=remote
--extra-config=kubelet.container-runtime-endpoint=/var/run/crio.sock
--extra-config=image-service-endpoint=/var/run/crio.sock
```

### CODE REVIEW

File: pkg/minikube/bootstrapper/kubeadm/kubeadm.go [hunk 1]
```diff
@@ -262,17 +302,35 @@ sudo systemctl start kubelet
 	return nil
 }
 
+func parseFeatureGates(featureGates string) (map[string]string, error) {
```
Comment (dlorenc) [inline pkg/minikube/bootstrapper/kubeadm/kubeadm.go]: I don't see this used outside of tests.

File: pkg/minikube/bootstrapper/kubeadm/templates.go [hunk 2]
```diff
@@ -30,21 +36,35 @@ networking:
...
-Environment="KUBELET_CGROUP_ARGS=--cgroup-driver=cgroupfs"
 ExecStart=
-ExecStart=/usr/bin/kubelet $KUBELET_KUBECONFIG_ARGS $KUBELET_SYSTEM_PODS_ARGS $KUBELET_DNS_ARGS $KUBELET_CADVISOR_ARGS $KUBELET_CGROUP_ARGS {{.ExtraOptions}}
+ExecStart=/usr/bin/kubelet {{.ExtraOptions}} {{if .FeatureGates}}--feature-gates={{.FeatureGates}}{{end}}
+
+[Install]
+{{installWants .ContainerRuntime}}
 `))
 
+func installWants(containerRuntime string) string {
```
Comment (dlorenc) [inline pkg/minikube/bootstrapper/kubeadm/templates.go]: Could you do this directly in the template with an if statement?

File: pkg/minikube/bootstrapper/kubeadm/kubeadm.go (hunk 1 above)
Comment (r2d4) [reply pkg/minikube/bootstrapper/kubeadm/kubeadm.go]: removed

File: pkg/minikube/bootstrapper/kubeadm/templates.go (hunk 2 above)
Comment (r2d4) [reply pkg/minikube/bootstrapper/kubeadm/templates.go]: done

**Pull request merged**